File: src/Tarea_3_Comparar/node.cpp
#include "node.h"
#include <sstream>
#include <iostream>
#include <string>


Node::Node(int pID)
{
    ID = pID;
    Hijo_Der = nullptr;
    Hijo_Izq = nullptr;

}
Node::Node(int pID, void *pValue){
    ID = pID;
    Value = pValue;
    Hijo_Der = nullptr;
    Hijo_Izq = nullptr;
           }


int Node:: getID() {
    return ID;
}

void Node::setHijo_Der(Node *pHijo_Der){
    Hijo_Der = pHijo_Der;
}

void Node::setHijo_Izq(Node *pHijo_Izq){
    Hijo_Izq = pHijo_Izq;
}

Node* Node::getHijo_Der(){
    return Hijo_Der;
}

Node* Node::getHijo_Izq(){
    return Hijo_Izq;
}

int Node::getBalanceo(){
    return balanceo;
}

void Node::setBalanceo(){
    //Falta implementar
}

void* Node::getValue(){
    return Value;
}

void Node::setArco(Node* pGrafo){
    arco = pGrafo;
}

Node* Node::getArco(){
    return arco;
}

void Node::setPeso(int pPeso){
    peso = pPeso;
}

int Node::getPeso(){
    return peso;
}

File: src/c_connection/Node.java
package c_connection;

import image_analysis.SampleImage;

public class Node {
	private SampleImage value;
	
	private Node arco;
	private int peso;
	
	private Node Hijo_Izq;
	private Node Hijo_Der;
	
	public SampleImage getValue() {
		return value;
	}

	public void setValue(SampleImage value) {
		this.value = value;
	}

	public Node getArco() {
		return arco;
	}

	public void setArco(Node arco) {
		this.arco = arco;
	}

	public int getPeso() {
		return peso;
	}

	public void setPeso(int peso) {
		this.peso = peso;
	}

	public Node getHijo_Izq() {
		return Hijo_Izq;
	}

	public void setHijo_Izq(Node hijo_Izq) {
		Hijo_Izq = hijo_Izq;
	}

	public Node getHijo_Der() {
		return Hijo_Der;
	}

	public void setHijo_Der(Node hijo_Der) {
		Hijo_Der = hijo_Der;
	}

	public Node(SampleImage pValue, int pPeso) {
		value = pValue;
		peso = pPeso;
	}

}
